core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -77,6 +77,16 @@ private[spark] class AppClient(
     private val registrationRetryThread =
       ThreadUtils.newDaemonSingleThreadScheduledExecutor("appclient-registration-retry-thread")
 
+    private val receiveAndReplyMaxPoolSize = conf.getInt(
+      "spark.appclient.receiveAndReply.maxThreads", 3)
+
+    private val receiveAndReplyThreadPool = new ThreadPoolExecutor(
```

Comment:
You can use `org.apache.spark.util.ThreadUtils.newDaemonCachedThreadPool` instead.